src/multivariate/hypersphericaluniform.jl

```diff
@@ -0,0 +1,28 @@
+"""
+Uniform distribution on the hyperspherical unit ball in ``d`` dimensions.
+"""
+struct HyperSphericalUniform <: ContinuousMultivariateDistribution
+    d :: Int
+end
+
+# basic properties
+Base.length(s::HyperSphericalUniform) = s.d
+Base.eltype(::HyperSphericalUniform) = Float32
+
+# sampling
+function _rand!(rng::AbstractRNG, ::HyperSphericalUniform, x::AbstractVector{T}) where T<:Real
+    x .= randn(rng, T, length(x))
+    normalize!(x)
+end
+
+function _rand(rng::AbstractRNG, ::HyperSphericalUniform, x::AbstractVector{T}) where T<:Real
+    normalize!(randn(rng, T, length(x)))
+end
+
+function StatsBase.entropy(s::HyperSphericalUniform)
+    return 1/2 * s.d * log(pi) - lgamma(s.d/2) + log(2)
```

Comment:
The suggestion promotes to Float64 as well since `halfd` is always a Float64. But I don't think that can be avoided, the result will always be `Float64` as there are no unknown types here. `lgamma` is deprecated, instead one should use `loggamma`.
```suggestion
    halfd = s.d / 2
    return halfd * logπ - loggamma(halfd) + logtwo
```

Reply:
One could probably use something like s.d / 2one(s.d), though I'm not sure how efficient that would be....

Reply:
Since the type of `s.d` is `Int`, this would not change the type of the output. Division by two is fine here.